azurerm/internal/services/mssql/resource_arm_mssql_database.go

```diff
@@ -0,0 +1,401 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMsSqlDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMsSqlDatabaseCreateUpdate,
+		Read:   resourceArmMsSqlDatabaseRead,
+		Update: resourceArmMsSqlDatabaseCreateUpdate,
+		Delete: resourceArmMsSqlDatabaseDelete,
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.MsSqlDatabaseID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateMsSqlDatabaseName,
+			},
+
+			"server_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.MsSqlServerID,
+			},
+
+			"auto_pause_delay_in_minutes": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validate.MsSqlDatabaseAutoPauseDelay,
+			},
+
+			//recovery is not support in version 2017-10-01-preview
+			"create_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.CreateModeCopy),
+					string(sql.CreateModeDefault),
+					string(sql.CreateModeOnlineSecondary),
+					string(sql.CreateModePointInTimeRestore),
+					string(sql.CreateModeRestore),
+					string(sql.CreateModeRestoreExternalBackup),
+					string(sql.CreateModeRestoreExternalBackupSecondary),
+					string(sql.CreateModeRestoreLongTermRetentionBackup),
+					string(sql.CreateModeSecondary),
+				}, false),
+			},
+
+			"collation": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.MsSqlDBCollation(),
+			},
+
+			"elastic_pool_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.MsSqlElasticPoolID,
+			},
+
+			"license_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.BasePrice),
+					string(sql.LicenseIncluded),
+				}, false),
+			},
+
+			"max_size_gb": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 1024),
```

Comment:
I think these values may be too restrictive?  Different vCore / DTU values have different limits, so I think a basic check is still the best option here, but update the docs to point the user at documentation resources for the value, such as: https://docs.microsoft.com/en-us/azure/sql-database/sql-database-service-tiers-general-purpose-business-critical
```suggestion
				ValidateFunc: validation.IntBetween(1, 4096),
```

Note: we may need to change it to `IntAtLeast(1)` in future if the service increases the upper boundary.